source/client/options_impl.cc

```diff
@@ -284,6 +294,16 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv) {
       throw MalformedArgvException(e.what());
     }
   }
+  if (!transport_socket.getValue().empty()) {
+    try {
+      envoy::api::v2::core::TransportSocket ts;
+      Envoy::MessageUtil::loadFromJson(transport_socket.getValue(), ts,
```

Comment:
Does loadFromJson also work on yaml?

Reply:
Good point: 

Two things are going on:
- this particular json parser tolerates not putting quotes around dictionary keys
- yaml's compact syntax seems to be the same as json except without quotes around dictionary keys

So I think we may actually be supporting both json and yaml with loadFromJson. Technically the documentation should say "json or compact yaml."

There's also generic loadFromFile that loads the whole file into memory to detect the format, but it takes an extra Api parameter that depends on a thread pool, and I didn't want to get involved with all that if loadFromJson happens to work. loadFromFile would be the correct thing to call if users needed to pass a multiline yaml string on the command line. (Or try loadFromJson, see if it throws an exception, and if so, try loadFromYaml...)